src/fsharp/FSharp.Core/array.fs

```diff
@@ -957,9 +957,13 @@ namespace Microsoft.FSharp.Collections
             accv
 
         [<CompiledName("Average")>]
-        let inline average      (array:_[]) = 
+        let inline average      (array:'T[]) = 
             checkNonNull "array" array
-            Seq.average array
+            if array.Length = 0 then invalidArg "array" LanguagePrimitives.ErrorStrings.InputArrayEmptyString
```

Comment:
Just show the performance times for Seq.average vs. the new Array,average for a range of inputs (many calls for a small array, fewer calls for larger arrays etc.)